stable/mariadb/templates/secrets.yaml

```diff
@@ -9,5 +10,10 @@ metadata:
     heritage: "{{ .Release.Service }}"
 type: Opaque
 data:
-  mariadb-root-password: {{ default "" .Values.mariadbRootPassword | b64enc | quote }}
+  {{- if .Values.mariadbRootPassword }}
+  mariadb-root-password: {{ .Values.mariadbRootPassword | b64enc | quote }}
+  {{- else }}
+  mariadb-root-password: {{ randAlphaNum 10 | b64enc | quote }}
+  {{- end }}
   mariadb-password: {{ default "" .Values.mariadbPassword | b64enc | quote }}
```

Comment:
It would be better to also make the user/application-password random by default instead of empty, then it would be secure by default.

Otherwise you would have to put a hardcoded password in values.yaml, which you wouldn't want to check into revision control etc. You can always read both the secrets from the application.